front/components/forms/signupform.tsx

```diff
@@ -1,7 +1,7 @@
 // a form for sign up
 
 import React from "react";
-import { translate } from "../../i18n/i18n";
+import { Translate, translate } from "../../i18n/i18n";
```

Comment:
Does the reload of `Translate` composant reload the `translate` function calls (like the function in this file are going to work as intended ? )

Reply:
The "Translate" component relies on a hook to retrieve the current language. Each time this changes, the component will be re-rendered, ie will recall the 'translate' function